server/src/DbtProfile.ts

```diff
@@ -12,12 +12,37 @@ export interface DbtProfile {
    * @param targetConfig target config specified in profiles.yml
    * @returns Err with missing field or Ok if profile is valid
    */
-  validateProfile(targetConfig: any): Result<void, string>;
+  validateProfile(targetConfig: TargetConfig): Result<void, string>;
 
   /**
    * Creates destination client according to dbt profile settings
    * @param profile profile specified in profiles.yml
    * @returns authenticated client or error string otherwise
    */
-  createClient(profile: any): Promise<Result<DbtDestinationClient, string>>;
+  createClient(profile: Required<TargetConfig>): Promise<Result<DbtDestinationClient, string>>;
+}
+
+export interface ProfileYaml {
+  target?: string;
+  outputs?: Partial<Record<string, TargetConfig>>;
+}
+
+export interface TargetConfig {
+  type?: DbtProfileType;
+  method?: string;
+  project?: string;
+  token?: string;
+  refresh_token?: string;
+  client_id?: string;
+  client_secret?: string;
+  keyfile_json?: {
+    private_key?: string;
+  };
+  keyfile?: string;
+  scopes?: string[];
+}
+
+export enum DbtProfileType {
+  BigQuery = 'bigquery',
+  Postgres = 'postgres',
```

Comment:
Linter complained about line `![...PROFILE_METHODS.keys()].find(t => t === type)` when `DbtProfileType` enum contains only one value
```
  104:48  error  Unnecessary conditional, both sides of the expression are literal values  @typescript-eslint/no-unnecessary-condition
```
But looks like we don't need this check because we have the same in `validateProfilesFile` method.